package.json

```diff
@@ -8,8 +8,8 @@
   },
   "scripts": {
     "test": "jest",
-    "dev": "webpack --config iXBRLViewerPlugin/viewer/webpack.dev.js",
-    "prod": "webpack --config iXBRLViewerPlugin/viewer/webpack.prod.js",
+    "dev": "VIEWER_VERSION=`git describe --tags --dirty`-dev webpack --config iXBRLViewerPlugin/viewer/webpack.dev.js",
```

Comment:
Does the plugin documentation need to be updated to make sure everyone includes this as part of their build? Or maybe it will just show an empty string?

Reply:
If the environment variable is not set, it results in an empty string, so there's nothing visible in the viewer.

Reply:
@paulwarren-wk FYI this only works correctly if the user has fetched all of the tags. So for me when I QA this prior to fetching everything:

![Screenshot 2023-03-07 at 4 40 03 PM](https://user-images.githubusercontent.com/17255946/223580835-2408c2ef-a091-4795-a2d5-6d8a658a2cb1.png)

v.s. after fetching everything

![Screenshot 2023-03-07 at 4 44 36 PM](https://user-images.githubusercontent.com/17255946/223581192-7650a04f-dd82-4f7e-a473-7b731dc5e909.png)

---

Its probably still an improvement but you might add `git fetch --tags` or something to your build command. I am also kind of iffy on if this will work well from forks who don't know about upstream, maybe thats an acceptable trade off though. I am fine with it as is, but lemme know if you don't want to change and I'll merge

Reply:
Interesting.  I wasn't aware that it was possible to do a `git fetch` without fetching tags, but it looks like there's a config option that makes not fetching tags the default.  Is this something you've selected?

I rather not add a `git fetch --tags` to the `dev` target, because it makes the build network-dependent, and even a no-op fetch can take a few seconds.  I'm also thinking that if someone has explicitly chosen to not fetch tags, then they may not want the build process to do so.

In your example, `1.1.38` may not be the most recent version number, but the `-70` does at least warn us that there have been quite a few changes since `1.1.38`.

> iffy on if this will work well from forks who don't know about upstream

I think it's only a problem if they start using their own tags that are confusable with our version numbers.  As long as they don't do that, then I think we'll just get an old version number and a large commit count, or whatever tags they choose to use on their fork.

Reply:
I'm doing `git fetch <YOUR_FORK` instead of `git fetch --tags` which I would say is fairly common for projects that have a lot of contributors

Reply:
> If the environment variable is not set, it results in an empty string, so there's nothing visible in the viewer.

Actually, there is an issue here which is that plugins will have their own webpack config, so unless they're updated to define `__VERSION__`, we'll get a `ReferenceError`.  I've just made a change to wrap the reference to `__VERSION__` in a `try`/`catch` to avoid this.

To get a version number showing in a plugin, the relevant webpack and build process will need to be updated to provide an appropriate definition of `__VERSION__`, so I think plugin authors are free to do that how they choose, although it would be useful to know what version of the viewer the plugin is built on - it should be possible to script `npm view ixbrl-viewer list` or similar when producing `__VERSION__` for a plugin.